db-engine/src/main/java/com/splicemachine/db/iapi/stats/FakeColumnStatisticsImpl.java

```diff
@@ -0,0 +1,160 @@
+ * See the GNU Affero General Public License for more details.
+ * You should have received a copy of the GNU Affero General Public License along with Splice Machine.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.splicemachine.db.iapi.stats;
+
+import com.splicemachine.db.iapi.types.DataValueDescriptor;
+import org.apache.log4j.Logger;
+
+import java.io.Externalizable;
+import java.io.IOException;
+import java.io.ObjectInput;
+import java.io.ObjectOutput;
+
+/**
+ * Created by yxia on 11/7/19.
+ */
+public class FakeColumnStatisticsImpl extends ColumnStatisticsImpl implements Externalizable {
+    private static Logger LOG=Logger.getLogger(FakeColumnStatisticsImpl.class);
+    private long nullCount;
+    private long totalCount;
+    private long cardinality;
+    private long rpv;
+
+    public FakeColumnStatisticsImpl() {
+    }
+
+    public FakeColumnStatisticsImpl(DataValueDescriptor dvd,
+                             long nullCount,
+                             long totalCount,
+                             long cardinality) {
+        this.dvd = dvd;
+        this.nullCount = nullCount;
+        this.totalCount = totalCount;
+        if (totalCount < nullCount)
+            this.totalCount = nullCount;
+        this.cardinality = cardinality;
+        if (cardinality > this.totalCount - nullCount + 1)
+            this.cardinality = this.totalCount - nullCount + 1;
+        rpv = (long) (((double)(this.totalCount - nullCount))/cardinality);
+
+    }
+
+    @Override
+    public Type getType() {
+        return Type.FAKE;
+    }
+
+    @Override
+    public DataValueDescriptor minValue() {
+        return null;
+    }
+
+    @Override
+    public DataValueDescriptor maxValue() {
+        return null;
+    }
+
+    @Override
+    public long totalCount() {
+        return totalCount;
+    }
+
+    @Override
+    public long nullCount() {
+        return nullCount;
+    }
+
+    @Override
+    public long notNullCount() {
+        long notNullCount = totalCount - nullCount;
+        if (notNullCount < 0)
+            notNullCount = 0;
+        return notNullCount;
+    }
+
+    @Override
+    public long cardinality() {
+        return cardinality;
+    }
+
+    @Override
+    public long selectivity(DataValueDescriptor element) {
+        // Use Null Data
+        if (element == null || element.isNull())
+            return nullCount;
+
+        if (rpv == -1)
+            rpv = (long) (((double)notNullCount())/cardinality);
+
+        return rpv;
+    }
+
+    @Override
+    public long rangeSelectivity(DataValueDescriptor start,DataValueDescriptor stop, boolean includeStart,boolean includeStop,boolean useExtrapolation) {
+        if ((start == null || start.isNull()) && (stop == null || stop.isNull()))
+            return nullCount;
+
+        if (start.equals(stop))
+            return rpv;
+
+        return (long)(0.1 * totalCount);
```

Comment:
Is this a default estimation from some other location?  If so, maybe use a static constant value so if one place gets updated, so does this place.

Reply:
Done. I added DEFAULT_RANGE_SELECTIVITY and a few others in SelectivityUtil and make it a constant.